src/main/java/taxi/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java

```diff
@@ -0,0 +1,144 @@
+            throw new DataProcessingException("Can not add new manufacturer"
+                    + manufacturer.getName(), e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String query = "SELECT * FROM manufacturers "
+                + "WHERE manufacturer_id = ? "
+                + "AND deleted = FALSE;";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                Optional<Manufacturer> newManufacturer = Optional
+                        .of(createManufacturer(resultSet));
+                preparedStatement.close();
+                resultSet.close();
+                return newManufacturer;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not find manufacturer with id: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String query = "SELECT * FROM manufacturers WHERE deleted = FALSE;";
+        List<Manufacturer> manufacturerList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                manufacturerList.add(createManufacturer(resultSet));
+            }
+            preparedStatement.close();
+            resultSet.close();
+        } catch (SQLException e) {
+            throw new DataProcessingException("No manufacturers found", e);
+        }
+        return manufacturerList;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String query = "UPDATE manufacturers"
+                + " SET manufacturer_name = ?, manufacturer_country = ?"
+                + " WHERE manufacturer_id = ?"
+                + " AND deleted = FALSE;";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            preparedStatement.setString(1, manufacturer.getName());
+            preparedStatement.setString(2, manufacturer.getCountry());
+            preparedStatement.setLong(3, manufacturer.getId());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            while (resultSet.next()) {
+                manufacturer = createManufacturer(resultSet);
+                preparedStatement.close();
+                resultSet.close();
+            }
```

Comment:
I'm not sure if it's needed as you pass fully formed manufacturer

Reply:
You are right - if we would not update the `manufacturer` we will catch an `exception`, otherwise the update was successful and we can return the input `manufacturer`.